contracts/Manager.sol

```diff
@@ -0,0 +1,413 @@
+    event ProposalExecuted(uint256 indexed id, address indexed proposer, address indexed executor);
+
+
+    // ============================ Constructor ===============================
+
+    /// Begins paused.
+    constructor(address vaultAddr, address rsvAddress, uint256 seigniorage_) public {
+        vault = IVault(vaultAddr);
+        rsv = IRSV(rsvAddress);
+        whitelist[_msgSender()] = true;
+        seigniorage = seigniorage_;
+        paused = true;
+        useWhitelist = true;
+    }
+
+    // ============================= Modifiers ================================
+
+    /// Modifies a function to run only when the contract is not paused.
+    modifier notPaused() {
+        _notPaused();
+        _;
+    }
+
+    /// Modifies a function to run only when the caller is on the whitelist, if it is enabled.
+    modifier onlyWhitelist() {
+        _onlyWhitelist();
+        _;
+    }
+
+    /// Modifies a function to run only when the caller is the operator account. 
+    modifier onlyOperator() {
+        _onlyOperator();
+        _;
+    }
+
+    function _notPaused() internal view {
+        require(!paused, "contract is paused");
+    }
+
+    function _onlyWhitelist() internal view {
+        if (useWhitelist) require(whitelist[_msgSender()], "not on whitelist");
+    }
+
+    function _onlyOperator() internal view {
+        require(_msgSender() == operator, "operator only");
+    }
+
+
+    // ============================= Externals ================================
+
+    /// Issue a quantity of RSV to the caller and deposit collateral tokens in the Vault.
+    function issue(uint256 _rsvQuantity) external notPaused onlyWhitelist {
+        _issue(_rsvQuantity);
+    }
+
+    // /// Issues the maximum amount of RSV to the caller based on their allowances.
+    // function issueMax() external notPaused onlyWhitelist {
+    //     uint256 max = _calculateMaxIssuable(_msgSender());
+    //     _issue(max);
+    // }
+
+    /// Redeem a quantity of RSV for collateral tokens. 
+    function redeem(uint256 _rsvQuantity) external notPaused onlyWhitelist {
+        _redeem(_rsvQuantity);
+    }
+
+    // /// Redeem `allowance` of RSV from the caller's account. 
+    // function redeemMax() external notPaused onlyWhitelist {
+    //     uint256 max = rsv.allowance(_msgSender(), address(this));
+    //     _redeem(max);
+    // }
+
+    /**
+     * Proposes an adjustment to the quantities of tokens in the Vault. Importantly, this type of
+     * proposal does not change token addresses. Therefore, if you want to introduce a new token,
+     * first use the other proposal type. 
+     */ 
+    function proposeQuantitiesAdjustment(
+        address[] calldata _tokens,
+        uint256[] calldata _amountsIn,
+        uint256[] calldata _amountsOut
+    ) 
+        external returns(uint256)
+    {
+        require(_amountsIn.length == _amountsOut.length, "quantities mismatched");
```

Comment:
Also `require(_tokens.length == _amountsIn.length, "token quantities mismatched");` ?

Reply:
Nice catch. Tokens was deleted and re-added and this was dropped.